drivers/staging/vc04_services/bcm2835-isp/bcm2835-v4l2-isp.c

```diff
@@ -229,16 +229,22 @@ static const struct bcm2835_isp_fmt *get_fmt(u32 mmal_fmt)
 
 static struct bcm2835_isp_fmt *find_format(struct v4l2_format *f,
 					   struct bcm2835_isp_node *node)
+{
+	return find_format_by_pix(f->fourcc, node);
+}
+
+static struct bcm2835_isp_fmt *find_format_by_pix(unsigned int pixelformat,
+						  struct bcm2835_isp_node *node)
 {
 	struct bcm2835_isp_fmt_list *fmts = &node->supported_fmts;
 	struct bcm2835_isp_fmt *fmt;
 	unsigned int i;
 
 	for (i = 0; i < fmts->num_entries; i++) {
 		fmt = &fmts->list[i];
-		if (fmt->fourcc == (node_is_stats(node) ?
-					    f->fmt.meta.dataformat :
-					    f->fmt.pix.pixelformat))
+		if (fmt->fourcc == node_is_stats(node) ?
+				   f->fmt.meta.dataformat :
+				   f->fmt.pix.pixelformat)
```

Comment:
Comment from the last version:

Haven't we lost the switch between f->fmt.meta.dataformat and f->fmt.pix.pixelformat totally?

I was expecting find_format to be
```return find_format_by_pix(node_is_stats(node) ? f->fmt.meta.dataformat : f->fmt.pix.pixelformat, node);```

Minor nit too that it isn't necessarily a pixelformat in here, it could be a dataformat. ```fourcc``` instead?
I haven't looked at how f->fourcc compares to these two parameters.
End quote.

This version you appear to be referencing f->fmt when f doesn't exist in find_format_by_pix

Reply:
Yeah sorry.  Got way too much going on right now...
Fixed.